source/heating_model/interface.cc

```diff
@@ -161,8 +308,56 @@ namespace aspect
       }
       prm.leave_subsection ();
 
+      prm.enter_subsection ("Model settings");
+      {
+        prm.declare_entry ("Include shear heating", "false",
+                           Patterns::Bool (),
+                           "Whether to include shear heating into the model or not. From a "
+                           "physical viewpoint, shear heating should always be used but may "
+                           "be undesirable when comparing results with known benchmarks that "
+                           "do not include this term in the temperature equation."
+                           "Warning: deprecated! Add 'shear heating' to the 'List of model "
+                           "names' instead.");
+        prm.declare_entry ("Include adiabatic heating", "false",
+                           Patterns::Bool (),
+                           "Whether to include adiabatic heating into the model or not. From a "
+                           "physical viewpoint, adiabatic heating should always be used but may "
+                           "be undesirable when comparing results with known benchmarks that "
+                           "do not include this term in the temperature equation."
+                           "Warning: deprecated! Add 'adiabatic heating' to the 'List of model "
+                           "names' instead.");
+        prm.declare_entry ("Include latent heat", "false",
+                           Patterns::Bool (),
+                           "Whether to include the generation of latent heat at phase transitions "
+                           "into the model or not. From a physical viewpoint, latent heat should "
+                           "always be used but may be undesirable when comparing results with known "
+                           "benchmarks that do not include this term in the temperature equation "
+                           "or when dealing with a model without phase transitions."
+                           "Warning: deprecated! Add 'latent heat' to the 'List of model "
+                           "names' instead.");
+      }
+      prm.leave_subsection ();
+
       std_cxx11::get<dim>(registered_plugins).declare_parameters (prm);
     }
+
+
+    HeatingModelOutputs::HeatingModelOutputs(const unsigned int n_points,
+                                             const unsigned int n_comp)
+      :
+      heating_source_terms(n_points),
+      lhs_latent_heat_terms(n_points)
+    {
```

Comment:
if 0 is a reasonable default (it is), you can skip setting them to 0 in Manager::evaluate()